src/snowflake/cli/plugins/snowpark/package_utils.py

```diff
@@ -68,165 +66,183 @@ def deduplicate_and_sort_reqs(
     """
     if requirements_file and package_name:
         raise ClickException(
             "Could not use package name and requirements file simultaneously"
         )
-
     if requirements_file and not requirements_file.exists():
         raise ClickException(f"File {requirements_file.path} does not exists.")
+    if perform_anaconda_check_for_dependencies and not anaconda:
+        raise ClickException(
+            "Cannot perform anaconda checks if anaconda channel is not specified."
+        )
 
-    with Venv() as v:
+    with Venv() as v, SecurePath.temporary_directory() as downloads_dir:
         if package_name:
             # This is a Windows workaround where use TemporaryDirectory instead of NamedTemporaryFile
-            tmp_requirements = SecurePath(v.directory.name) / "requirements.txt"
-            tmp_requirements.write_text(str(package_name))
-            requirements_file = tmp_requirements
-
-        assert type(requirements_file) is SecurePath
-        pip_install_result = v.pip_install(str(requirements_file.path))
-        dependencies = v.get_package_dependencies(str(requirements_file.path))
+            requirements_file = SecurePath(v.directory.name) / "requirements.txt"
+            requirements_file.write_text(str(package_name))
 
-        if pip_install_result != 0:
-            log.info(pip_failed_msg.format(pip_install_result))
+        pip_wheel_result = v.pip_wheel(
+            requirements_file.path, download_dir=downloads_dir.path, index_url=index_url  # type: ignore
+        )
+        if pip_wheel_result != 0:
+            log.info(pip_failed_log_msg, pip_wheel_result)
             return False, None
 
-        if perform_anaconda_check:
-            log.info("Checking for dependencies available in Anaconda...")
-            dependency_requirements = [dep.requirement for dep in dependencies]
-            log.info(
-                "Downloaded packages: %s",
-                ",".join([d.name for d in dependency_requirements]),
-            )
-            second_chance_results: SplitRequirements = anaconda.parse_anaconda_packages(
-                packages=dependency_requirements
-            )
-
-            if len(second_chance_results.snowflake) > 0:
-                log.info(second_chance_msg.format(second_chance_results.snowflake))
-            else:
-                log.info("None of the package dependencies were found on Anaconda")
+        dependencies = v.get_package_dependencies(
+            requirements_file, downloads_dir=downloads_dir.path
+        )
+        dependency_requirements = [d.requirement for d in dependencies]
 
-        dependencies_to_be_packed = _get_dependencies_not_avaiable_in_conda(
-            dependencies,
-            second_chance_results.snowflake if second_chance_results else None,
+        log.info(
+            "Downloaded packages: %s",
+            ",".join([d.name for d in dependency_requirements]),
```

Comment:
```suggestion
        log.info(
            "Downloaded packages: %s",
            ", ".join([d.name for d in dependency_requirements]),
```
To make it more human-readable

Reply:
fixed